ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -66,48 +156,89 @@ public BlobId(String id, ClusterMap clusterMap) throws IOException {
    * @param clusterMap of the cluster that the blob id belongs to
    * @throws IOException
    */
-  public BlobId(DataInputStream stream, ClusterMap clusterMap) throws IOException {
-    this(stream, clusterMap, false);
+  public static BlobId fromDataInputStream(DataInputStream stream, ClusterMap clusterMap) throws IOException {
+    return new BlobId(stream, clusterMap, false);
   }
 
   /**
-   * Re-constructs existing blobId by deserializing from data input stream. This constructor includes an optional check
-   * that the stream has no more available bytes after reading.
-   *
-   * @param stream from which to deserialize the blobid
-   * @param clusterMap of the cluster that the blob id belongs to
-   * @param ensureFullyRead {@code true} if the stream should have no more available bytes after deserializing the blob
-   *                        ID.
-   * @throws IOException
+   * Size of blob id when it is serialized into bytes.
+   * @return The byte count of the serialized blob id.
    */
-  private BlobId(DataInputStream stream, ClusterMap clusterMap, boolean ensureFullyRead) throws IOException {
-    this.version = stream.readShort();
-    if (version == 1) {
-      partitionId = clusterMap.getPartitionIdFromStream(stream);
-      if (partitionId == null) {
-        throw new IllegalArgumentException("Partition ID cannot be null");
-      }
-      uuid = Utils.readIntString(stream);
-      if (ensureFullyRead && stream.read() != -1) {
-        throw new IllegalArgumentException("Stream should have no more available bytes to read");
-      }
-    } else {
-      throw new IllegalArgumentException("version " + version + " not supported for blob id");
-    }
-  }
-
   public short sizeInBytes() {
-    return (short) (Version_Size_In_Bytes + partitionId.getBytes().length + UUID_Size_In_Bytes + uuid.length());
+    switch (version) {
+      case BLOB_ID_V1:
+        return (short) (VERSION_SIZE_IN_BYTES + partitionId.getBytes().length + UUID_SIZE_IN_BYTES + uuid.length());
+      case BLOB_ID_V2:
+        return (short) (VERSION_SIZE_IN_BYTES + FLAG_SIZE_IN_BYTES + DATACENTER_ID_SIZE_IN_BYTES
+            + ACCOUNT_ID_SIZE_IN_BYTES + CONTAINER_ID_SIZE_IN_BYTES + partitionId.getBytes().length + UUID_SIZE_IN_BYTES
+            + uuid.length());
+      default:
+        throw new IllegalArgumentException("blob id version=" + version + " not supported");
+    }
   }
 
+  /**
+   * Gets the {@link PartitionId} this blob belongs to.
+   * @return The {@link PartitionId}.
+   */
   public PartitionId getPartition() {
     return partitionId;
   }
 
+  /**
+   * Gets the id of the {@link Account} who created this blob. If this information was not available when the
+   * blob id was formed, it will return -1.
```

Comment:
-1 only in case of older versioned blobId. May be good to call it out explicitly.